### DISCUSSION TITLE
is it possible to run mfsdirinfo remotely?

### DISCUSSION THREAD

**Discussion by eleaner** [moosefs/moosefs#503] [General]
I don't seem to find an option to specify the server or port.

I am trying to find out which of my moose folders started growing, and I am trying to use a docker container to send some data to influxdb. connecting remotely to master would be a perfect solution.
I can do simple ls, but mfsdirinfo has much more interesting info.

when trying to run mfsdirinfo from within the container (moosefs path mapped), I get
`can't connect to master: EINPROGRESS (Operation now in progress)`
it looks like mfsdirinfo might be actually querying the mount process

**Comment by eleaner** [comment]
@chogata 
please, any info?

**Comment by chogata** [comment]
Yes, mfsclient acts as proxy for tools, mfsdirinfo included.

**Reply by eleaner** [reply]
@chogata 
Would it make sense to expose the port of mfsclient and allow querying this way rather than querying the process?
Or, in other words, is there another way to run mfsdirinfo from inside the docker container without having mfsclient running there too?

**Reply by chogata** [reply]
This was how it worked in version 1.0 :) We changed that model - for a reason. The client keeps an open connection to the master, a tool opens a new connection every time it is run. Users killed the master when they run big shell loops calling tools many times inside. Of course, one could supply a tool that connects directly to the master and just comes with warnings, the problem is - users rarely read the warnings and then they ask "why is the tool killing my master?". So this change was made for safety/efficiency purposes.
It's not like the client port is closed in the master (it can't be, the client needs it ;) ), it is theoretically possible to write a tool that will connect directly and do what mfsdirinfo is doing, we are just not bundling such a tool with mfs, for reasons I just described. I can see where it might be a problem with containers, but if we were to change our model, it would be to special files with direct mode or something like that, not to individual connections of tools with the master.

**Reply by eleaner** [reply]
thank you for the explanation
much appreciated!
